Multiple client address deidentification

kairstenfay opened the pull request:
This version of the geocoding program accepts multiple command line arguments for on-the-fly configuration. Known collaborators (such as UW), can pass their institute acronym to the 'institute' option to load saved configuration.

joverlee521 commented on src/address_to_census_tract.py:
```diff
@@ -0,0 +1,374 @@
+
+@click.command()
+@click.argument('filepath', required=True, type=click.Path(exists=True))
+@click.option('-i', '--institute', default='default', 
```
Would this option be better as a choice option so that users know the exact acronym they need to enter?
https://click.palletsprojects.com/en/7.x/options/#choice-options

kairstenfay replied:
Good find!

kairstenfay committed:
Update gitignore

Ignore cached compiled Python modules

kairstenfay committed:
Add test CSV data

kairstenfay committed:
Use SmartyStreets geocoding API for addresses

These changes swapped Google's geocoding API for SmartyStreets' via its 
Python SDK. Using this service requires an authentication ID and token not tracked
through git that users need to add to their environment variables. 

The program currently works by passing a free text address via the `street`
parameter to the SmartyStreets  US Street API. If no match is found, the program 
re-tries the same address by passing it to the `text` parameter of the 
SmartyStreets US Extract API. The data is printed to stdout with additional
information on successful matches containing standardized address, latitude,
longitude, and census tract.

Currently, this program can be run with the following command.
`./address_to_census_tract.py {filepath}` where `filepath` has an extension of
either `.json` or `.csv`. See the example data sets in `data/test` for column or
category naming conventions.

kairstenfay committed:
Squash

kairstenfay committed:
Merge branch 'smarty-streets-address-standardization' into multiple-client-address-deidentification

kairstenfay committed:
Checkpoint
